pkg/cloud/awsprovider.go

```diff
@@ -1537,107 +1529,91 @@ func (a *AWS) GetDisks() ([]byte, error) {
-		svc = athena.New(s, &aws.Config{
-			Region:      region,
-			Credentials: creds,
-		})
-	}
-
-	var e athena.StartQueryExecutionInput
 
-	var r athena.ResultConfiguration
-	r.SetOutputLocation(resultsBucket)
-	e.SetResultConfiguration(&r)
+	queryExecutionCtx := &athenaTypes.QueryExecutionContext{
+		Database: awsSDK.String(awsAthenaInfo.AthenaDatabase),
+	}
 
-	e.SetQueryString(query)
-	var q athena.QueryExecutionContext
-	q.SetDatabase(database)
-	e.SetQueryExecutionContext(&q)
+	resultConfiguration := &athenaTypes.ResultConfiguration{
+		OutputLocation: awsSDK.String(awsAthenaInfo.AthenaBucketName),
+	}
+	startQueryExecutionInput := &athena.StartQueryExecutionInput{
+		QueryString:           awsSDK.String(query),
+		QueryExecutionContext: queryExecutionCtx,
+		ResultConfiguration:   resultConfiguration,
+	}
 
-	res, err := svc.StartQueryExecution(&e)
+	// Create Athena Client
+	cfg, err := awsAthenaInfo.CreateConfig()
 	if err != nil {
-		return err
+		log.Errorf("Could not retrieve Athena Configuration: %s", err.Error())
 	}
+	cli := athena.NewFromConfig(cfg)
 
-	klog.V(2).Infof("StartQueryExecution result:")
-	klog.V(2).Infof(res.GoString())
-
-	var qri athena.GetQueryExecutionInput
-	qri.SetQueryExecutionId(*res.QueryExecutionId)
-
-	var qrop *athena.GetQueryExecutionOutput
-	duration := time.Duration(2) * time.Second // Pause for 2 seconds
-
-	for {
-		qrop, err = svc.GetQueryExecution(&qri)
-		if err != nil {
-			return err
-		}
-		if *qrop.QueryExecution.Status.State != "RUNNING" && *qrop.QueryExecution.Status.State != "QUEUED" {
-			break
-		}
-		time.Sleep(duration)
+	// Query Athena
+	startQueryExecutionOutput, err := cli.StartQueryExecution(ctx, startQueryExecutionInput)
+	if err != nil {
+		log.Errorf(err.Error())
+	}
+	waitForQueryToComplete(ctx, cli, startQueryExecutionOutput.QueryExecutionId)
+	queryResultsInput := &athena.GetQueryResultsInput{
+		QueryExecutionId: startQueryExecutionOutput.QueryExecutionId,
 	}
-	if *qrop.QueryExecution.Status.State == "SUCCEEDED" {
+	getQueryResultsPaginator := athena.NewGetQueryResultsPaginator(cli, queryResultsInput)
+	for getQueryResultsPaginator.HasMorePages() {
+		pg, _ := getQueryResultsPaginator.NextPage(ctx)
+		fn(pg)
+	}
+	return nil
+}
 
-		var ip athena.GetQueryResultsInput
-		ip.SetQueryExecutionId(*res.QueryExecutionId)
-		err = svc.GetQueryResultsPages(&ip, fn)
-		if err != nil {
-			return fmt.Errorf("queryAthenaPaginated: error getting query resultsPages from athena service %s", err)
+func waitForQueryToComplete(ctx context.Context, client *athena.Client, queryExecutionID *string) {
+	inp := &athena.GetQueryExecutionInput{
+		QueryExecutionId: queryExecutionID,
+	}
+	isQueryStillRunning := true
+	for isQueryStillRunning {
+		qe, _ := client.GetQueryExecution(ctx, inp)
+		if qe.QueryExecution.Status.State == "SUCCEEDED" {
+			isQueryStillRunning = false
 		}
-		return nil
-	} else {
-		return fmt.Errorf("No results available for %s", query)
+		time.Sleep(10 * time.Second)
```

Comment:
Do we really want to wait for 10 seconds?

Reply:
I definitely think 10seconds is a long time to wait between polls. Was 2 seconds causing problems?

Reply:
I lifted this directly from @AjayTripathy's branch, 10 seconds does seem a little long. I think 2 sounds more appropriate

Reply:
updated to 2